### PR TITLE
MP4 tagger

### PR DESCRIPTION
**Pull request by cbj4074**
As has been done for the other audio formats, add support for ALAC (MP4) files.

### CODE REVIEW

File: src/Alac/AlacManipulatorCreator.php [hunk 1]
```diff
@@ -0,0 +1,24 @@
+<?php
+
+namespace IndieHD\AudioManipulator\Alac;
+
+use IndieHD\AudioManipulator\Converting\ConverterInterface;
+use IndieHD\AudioManipulator\Tagging\TaggerInterface;
+
+class AlacManipulatorCreator implements AlacManipulatorCreatorInterface
```
Comment (cbj4074) [inline src/Alac/AlacManipulatorCreator.php:8]: While this class's purpose may not be obvious, the intention is to be able to resolve `AlacManipulator` from the container, albeit indirectly (i.e., through this "proxy" class), given that its constructor requires a primitive argument (in this case, a string that represents a file path).

The use-case is explained very clearly at (this is not my post, FWIW):

https://stackoverflow.com/questions/46084709/how-to-place-a-class-that-requires-a-variable-parameter-into-dependency-injectio

If there's a better means by which to accomplish this, I'm all ears.

File: src/Alac/AlacTagger.php [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+        $this->validator = $validator;
+
+        // This option is specific to the tag READER (the WRITER has its own,
+        // separate encoding setting).
+
+        $this->getid3->setOption(['encoding' => 'UTF-8']);
+
+        // TODO Make the log location configurable.
+
+        $fileHandler = new StreamHandler(
```
Comment (cbj4074) [inline src/Alac/AlacTagger.php:50]: Shame on me. I'll fix this shortly.

Comment (cbj4074) [reply src/Alac/AlacTagger.php:50]: Actually, upon further consideration, the scenario here is the same as I [described elsewhere in this PR](https://github.com/indiehd/audio-manipulator/pull/9/files#r280066346), although, I didn't create the proxy class as I did in the aforementioned case.

Before I change anything here, let's work through the other discussion, as the same solution applies here.

Comment (cbj4074) [reply src/Alac/AlacTagger.php:50]: I know how I'm going to fix this, but it's beyond the scope of this PR. Let's table this for now and I'll fix it in a separate PR.

Comment (cbj4074) [reply src/Alac/AlacTagger.php:50]: Created #13 to track this.

**Review by poppabear8883 — APPROVED** [review]

**Pull request merged**